codegen/module.go

```diff
@@ -627,7 +628,7 @@ func (system *ModuleSystem) ResolveModules(
 	return classArrayModuleMap, nil
 }
 
-func (system *ModuleSystem) getInstanceFunc(baseDirectory string, className string, packageRoot string, targetGenDir string, defaultDependencies []ModuleDependency, moduleDirectoryGlob string) func(moduleDirAbsInf interface{}) (interface{}, error) {
+func (system *ModuleSystem) getInstanceFunc(baseDirectory string, className string, packageRoot string, targetGenDir string, defaultDependencies []ModuleDependency, moduleDirectoryGlob string, enableCustomInitialisation bool) func(moduleDirAbsInf interface{}) (interface{}, error) {
```

Comment:
The number of params is getting out of hand. Can we make this into a `params` struct?